content/en/blog/2022-05-05-introducing-kluctl-and-flux.md

```diff
@@ -0,0 +1,83 @@
+database is required to make your application work, you can manage the redis Helm Release and your application in the
+same deployment, including the necessary configuration to let them talk to each other.
+
+To see how different a Kluctl deployment is compared to classic Kustomize/Helm + Flux, you can compare the
+[flux2-kustomize-helm-example](https://github.com/fluxcd/flux2-kustomize-helm-example) and the
+[Kluctl Microservices Demo](https://github.com/kluctl/kluctl-examples/tree/main/microservices-demo/3-templating-and-multi-env)
+([here]({{< ref "docs/guides/tutorials/microservices-demo">}}) is tutorial for the demo).
+
+## Native multi-env support
+Kluctl allows you to natively create deployment projects that can be deployed multiple times to different
+environments/targets. You can for example have one target that is solely meant for `local` (e.g. Kind based) deployments,
+one that targets the `test` environment and one for `prod`. You can then use templating to influence deployments in whatever
+way you like. For example, you could change the `local` target to set all replicas to 1 and omit resource hungry
+support applications.
+
+This is possible in plain Kustomize as well, but requires you to solve it without the concept of
+[targets]({{< ref "docs/reference/kluctl-project/targets" >}}) and without templating. In Kustomize, multi-env
+deployments must be solved with [overlays](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/kustomization/#bases-and-overlays),
+which does not necessary align with how you prefer your project structure.
+
+## Mix local DevOps and GitOps
+The core idea of GitOps is that Git becomes the single source of truth for the desired cluster state. This is something
+that is extremely valuable with many advantages compared to other approaches. There are however still situations
+where diverging from GitOps is very valuable as well.
+
+For example, when you start a new deployment project, you're usually in a state of frequent changes inside the deployment
+project. These frequent changes need frequent deployments and testing until you get to a point where things are stable
+enough. If you're forced to adhere to GitOps in that situation, you end up with very noisy Git histories and plenty
+of trial-and-error deployment cycles. This is, at least in our opinion and experience, a major productivity killer
+without any advantages.
+
+With Kluctl, you can start developing locally and deploying from your local machine, with the guarantee that what you
+see is what will also happen later when GitOps is introduced for the same deployment. When you're "ready", you can then
+commit and push to Git and create the appropriate `KluctlDeployment` objects and then let GitOps/Flux take over.
+
+You can also use dedicated targets for development purposes and only deploy to them from your local machine, while
+other targets are deployed via GitOps/Flux.
```

Comment:
Kluctl offers you, as long as you have access to the cluster, the possibility to compare or deploy your local customizations, or only parts of if, against all environments at any time. This is what is special about Kluctl, as it has the CLI functionality in its DNA.

Reply:
`That brings trust to your deployments, and is missing in many a GitOps tool out there`
maybe to aggressive ;)

Reply:
Flux can actually do this in a limited fashion :) See [here](https://fluxcd.io/docs/cmd/flux_diff_kustomization/).